frontend/public/components/droppable-edit-yaml.tsx

```diff
@@ -0,0 +1,68 @@
+import withDragDropContext from './utils/drag-drop-context';
+import * as React from 'react';
+import {EditYAML} from './edit-yaml';
+import { NativeTypes } from 'react-dnd-html5-backend';
+// eslint-disable-next-line no-unused-vars
+import { DropTarget } from 'react-dnd';
+
+const boxTarget = {
+  drop(props, monitor) {
+    if (props.onDrop && monitor.isOver()) {
+      props.onDrop(props, monitor);
+    }
+  },
+};
+
+const EditYAMLComponent = DropTarget(NativeTypes.FILE, boxTarget, (connectObj, monitor) => ({
+  connectDropTarget: connectObj.dropTarget(),
+  isOver: monitor.isOver(),
+  canDrop: monitor.canDrop(),
+}))(EditYAML);
+
+
+export const DroppableEditYAML = withDragDropContext(class DroppableEditYAML extends React.Component<DroppableEditYAMLProps, DroppableEditYAMLState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      obj: this.props.obj || '',
+    };
+    this.handleFileDrop = this.handleFileDrop.bind(this);
+  }
+  handleFileDrop(item, monitor) {
+    if (!monitor) {
+      return;
+    }
+    const file = monitor.getItem().files[0];
+    // limit size size uploading to 1 mb
+    if (file.size <= 1048576) {
```

Comment:
I think we should align with https://github.com/openshift/console/blob/master/frontend/public/components/utils/file-input.tsx#L9-L10